test/check/do_single_expression_test.exs

```diff
@@ -0,0 +1,115 @@
+defmodule CompassCredoPlugin.Check.DoSingleExpressionTest do
+  use Credo.Test.Case
+
+  alias CompassCredoPlugin.Check.DoSingleExpression
+
+  describe "given valid functions, if statements and various declariations" do
+    test "does NOT report an issue" do
+      module_source_code = """
+      defmodule CredoSampleModule do
+        my_map = %{
+          a: 1
+        }
+
+        def some_function() do
+           a = 5 + 7
+           a + 5
+        end
+
+        def get_response(), do: {:ok, :response}
+
+        if some_condition,
+          do: :ok
+
+        unless 1 > 2, do: :ok
+
+        def some_other_function, do: :ok
+
+        defp another_function(), do: :ok
+      end
+      """
+
+      module_source_code
+      |> to_source_file()
+      |> run_check(DoSingleExpression)
+      |> refute_issues()
+    end
+  end
+
+  describe "given all the functions and if statements are INVALID" do
+    test "reports an issue on all instances" do
+      module_source_code = """
+      defmodule CredoSampleModule do
+        def some_function() do
+           a = 5 + 7
+        end
+
+        if some_condition do
+          :ok
+        end
+
+        unless 1 > 2 do
+          :ok
+        end
```

Comment:
It looks like it is valid syntax, but usually, we are using `if/unless` inside `def`, not inside `defmodule`.

Can we adjust tests of `if/unless` for something like:
```
def some_function() do
   unless 1 > 2 do
      :ok
    end
end
```

Reply:
Nice catch! :)

Reply:
Changed here @Nihisil [0cc4f16](https://github.com/nimblehq/compass-credo-plugin/pull/14/commits/0cc4f16f69073ce7fb342a96dcae847a2a63aab3)